django_ipam/base/models.py

```diff
@@ -34,14 +34,27 @@ def clean(self):
             if self.id != subnet["id"] and ip_network(self.subnet).overlaps(subnet["subnet"]):
                 raise ValidationError({'subnet': _('Subnet overlaps with %s') % (subnet["subnet"])})
 
-    def request_ip(self):
+    def get_first_available_ip(self):
         subnet = get_object_or_404(swapper.load_model("django_ipam", "Subnet"), pk=self.id)
         ipaddress_set = [ipaddr.ip_address for ipaddr in subnet.ipaddress_set.all()]
         for host in subnet.subnet.hosts():
             if str(host) not in ipaddress_set:
                 return str(host)
         return None
 
+    def request_ip(self, options={}):
```

Comment:
change to `options=None`, then in the code add:
```python
if options is None:
    options = {}
```
never use a mutable object as default value, very weird things can happen because that object is loaded during start time and the object lives until the program shuts down, that means that every modification made to it will modify the default value as well during the life of the program.